### DISCUSSION TITLE
joinedload Query option ignored with session.execute

### DISCUSSION THREAD

**Discussion by martijnthe** [sqlalchemy/sqlalchemy#11118] [Usage Questions] [answered]
When using `session.execute()` directly on a `Query` object (without `.statement`), the query is executed (nice!) but loader options seem to be ignored  (footgun!). 

Adding `.statement` "fixes" this, but it's easy to forget/miss.
What's the rationale for executing a `Query` w/o the loader options?

Minimal repro example:
```
from sqlalchemy import Column, create_engine, ForeignKey, BigInteger, inspect
from sqlalchemy.orm import sessionmaker, relationship, declarative_base, Mapped, joinedload

engine = create_engine('DB_URL', echo=True, future=True)
Session = sessionmaker(engine, expire_on_commit=False)

Base = declarative_base()


class A(Base):
    __tablename__ = "a"

    id: Mapped[int] = Column(BigInteger, primary_key=True)

    b_id: Mapped[int] = Column(BigInteger, ForeignKey("b.id"))
    b = relationship("B", uselist=False)


class B(Base):
    __tablename__ = "b"

    id: Mapped[int] = Column(BigInteger, primary_key=True)


Base.metadata.create_all(engine)


with Session() as session:
    with session.begin():
        b = B()
        session.add(b)
        session.flush()
        a = A(b=b)
        session.add(a)
        session.commit()


with Session() as session:
    with session.begin():
        q = session.query(A).options(joinedload(A.b))

        # Eager loading options are ignored -- note: missing q.statement:
        rv = session.execute(q).all()
        # 'A's are loaded OK:
        first_a = rv[0][0]

        ins = inspect(first_a)
        assert "b" not in ins.unloaded, ins.unloaded  # FAILS!
```

**Comment by zzzeek** [comment, marked as answer]
> When using session.execute() directly on a Query object (without .statement), the query is executed (nice!) but loader options seem to be ignored (footgun!).

You shouldn't really be doing that.   if you have a Query, just iterate it or call all().  if you want to use session.execute(), change the call to `q = session.query(Class)` to `select(Class)`.

> What's the rationale for executing a Query w/o the loader options?

it's not intentional API, it's something that "works" and I think we've wanted to deprecate it but perhaps we got pushback or something, @CaselIT do you recall

**Reply by CaselIT** [reply]
I remember we talked about it, but we don't seem to have an issue for it. That's fine for me, but probably we keep it deprecated also on 2.1?

**Reply by zzzeek** [reply]
yeah, i'm not in any hurry to remove this, just in even less of a hurry to....support it :)

**Reply by martijnthe** [reply]
@zzzeek thank you for the lightning fast response!

What about doing `session.execute(query.statement)`? Is that sane?

**Reply by zzzeek** [reply]
if the options are transferred over, then it's fine sure.  not sure if the opts are transferred for that

**Reply by CaselIT** [reply]
added https://github.com/sqlalchemy/sqlalchemy/issues/11139

**Comment by martijnthe** [comment]
Thanks, I'll close as Resolved!
